<launch>
  <arg name="robot_name"                default="sgr532"/>
  <arg name="robot_model"               default="$(arg robot_name)"/>
  <arg name="arm_velocity"              default="2048"/>
  <arg name="arm_acceleration"          default="10"/>
  <arg name="joint_pub_gui"             default="false"/>
  <arg name="use_joint_pub"             default="true"/>
  <arg name="just_rviz_control"         default="false"/> 
  
  <!-- RViz + URDF -->
  <include file="$(find sagittarius_descriptions)/launch/description.launch">
    <arg name="use_joint_pub" value="$(arg use_joint_pub)"/>
    <arg name="joint_pub_gui" value="$(arg joint_pub_gui)"/>
  </include>

  <!-- Arm Driver Node -->
  <node name="sdk_sagittarius_arm"
        pkg="sdk_sagittarius_arm"
        type="sdk_sagittarius_arm"
        output="screen"
        respawn="false"
        ns="$(arg robot_name)">
    
    <param name="robot_name" value="$(arg robot_name)"/>
    <param name="robot_model" value="$(arg robot_model)"/>
    <param name="arm_velocity" value="$(arg arm_velocity)"/>
    <param name="arm_acceleration" value="$(arg arm_acceleration)"/>
    <param name="servo_torque1" value="300"/>
    <param name="servo_torque2" value="300"/>
    <param name="servo_torque3" value="300"/>
    <param name="servo_torque4" value="200"/>
    <param name="servo_torque5" value="300"/>
    <param name="servo_torque6" value="200"/>
    <param name="servo_torque7" value="100"/>
    <param name="range_min" type="double" value="0.01" />
    <param name="range_max" type="double" value="30.0" />
    <param name="serialname" type="string" value="/dev/sagittarius" />
    <param name="baudrate" type="string" value="1000000" />
    <param name="timelimit" type="int" value="5000" />
    <param name="servo_configs" type="string" value="$(find sdk_sagittarius_arm)/cfg/"/>
    <param name="just_rviz_control" type="bool" value="$(arg just_rviz_control)"/>
    <param name="servo_control_trajectory" type="bool" value="false" />
  </node>

  <!-- Pre Pick Pose Node -->
    <node name="send_pre_pick_pose"
        pkg="sagittarius_control"
        type="send_pre_pick_pose.py"
        output="screen">
    <param name="config_path" value="$(find sagittarius_control)/config/pre_pick_pose.yaml"/>
</node>
</launch>
